### ISSUE TITLE
Cant run anything at all. Possible numpy incompatibility?

### ISSUE THREAD

**Issue by adityasingh8010** [nikhil-ravi/LeetScrape#12]
Traceback (most recent call last):
  File "C:\Users\adity\OneDrive - Louisiana State University\Documents\leetscrape_test\test1.py", line 1, in <module>
    from leetscrape import GetQuestionsList
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\__init__.py", line 2, in <module>
    from .generate_code_stub import GenerateCodeStub
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\generate_code_stub.py", line 9, in <module>
    from .question import GetQuestion
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\question.py", line 5, in <module>
    import pandas as pd
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\__init__.py", line 22, in <module>
    from pandas.compat import is_numpy_dev as _is_numpy_dev  # pyright: ignore # noqa:F401      
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\compat\__init__.py", line 18, in <module>
    from pandas.compat.numpy import (
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\compat\numpy\__init__.py", line 4, in <module>
    from pandas.util.version import Version
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\util\__init__.py", line 2, in <module>
    from pandas.util._decorators import (  # noqa:F401
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\util\_decorators.py", line 14, in <module>
    from pandas._libs.properties import cache_readonly
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\_libs\__init__.py", line 13, in <module>
    from pandas._libs.interval import Interval
  File "pandas\_libs\interval.pyx", line 1, in init pandas._libs.interval
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject

(myenv) C:\Users\adity\OneDrive - Louisiana State University\Documents\leetscrape_test>python test1.py
Traceback (most recent call last):
  File "C:\Users\adity\OneDrive - Louisiana State University\Documents\leetscrape_test\test1.py", line 1, in <module>
    from leetscrape import GetQuestion
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\__init__.py", line 2, in <module>
    from .generate_code_stub import GenerateCodeStub
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\generate_code_stub.py", line 9, in <module>
    from .question import GetQuestion
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\leetscrape\question.py", line 5, in <module>
    import pandas as pd
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\__init__.py", line 22, in <module>
    from pandas.compat import is_numpy_dev as _is_numpy_dev  # pyright: ignore # noqa:F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\compat\__init__.py", line 18, in <module>
    from pandas.compat.numpy import (
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\compat\numpy\__init__.py", line 4, in <module>
    from pandas.util.version import Version
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\util\__init__.py", line 2, in <module>
    from pandas.util._decorators import (  # noqa:F401
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\util\_decorators.py", line 14, in <module>
    from pandas._libs.properties import cache_readonly
  File "C:\Users\adity\miniconda3\envs\myenv\Lib\site-packages\pandas\_libs\__init__.py", line 13, in <module>
    from pandas._libs.interval import Interval
  File "pandas\_libs\interval.pyx", line 1, in init pandas._libs.interval
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject

**Comment by kushagrchitkar** [comment]
This error happens because of a version mismatch between NumPy and Pandas.

The leetscrape package requires pandas ^1.5.2.

But during installation, you likely got NumPy 2.x, which is not compatible with Pandas 1.5.x.

Pandas 1.5.x was built against NumPy 1.x, and NumPy 2.x changed some internals (like the dtype structure), which breaks compatibility.

TL;DR
pip uninstall pandas numpy
pip install "numpy<2.0" "pandas>=1.5.2,<2.0"
